app/models/immunisation_import_row.rb

```diff
@@ -380,6 +386,13 @@ def programmes_by_name
 
   def offline_recording? = session_id.present?
 
+  def imms_api_record?
+    uuid.present? &&
+      VaccinationRecord.find_by!(
+        uuid: uuid.to_s
+      ).sourced_from_nhs_immunisations_api?
```

Comment:
```suggestion
      VaccinationRecord.sourced_from_nhs_immunisations_api.exists?(
        uuid: uuid.to_s
      )
```

I think we should try and stick with `exists?` in this class to improve performance. `EXISTS` will be much faster as Rails won't need to select all the columns from the table and create a `VaccinationRecord` instance for it, instead we'll get a boolean straight from the database. Especially since this method is now going to be called for every row in the file.